viewer-prototype/src/browser/trace-viewer/components/utils/tooltip-component.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+
+type TooltipProps = {
+    tooltip: { [key: string]: string };
+}
+
+export class TooltipComponent extends React.Component<TooltipProps> {
+    constructor(props: TooltipProps) {
+        super(props); 
+    }
+
+    tooltipRef = React.createRef<any>();
+
+
+    renderTooltip() {
+        const tooltipArray: any[] = [];
+        if (this.props.tooltip) {
+            const keys = Object.keys(this.props.tooltip);
+            keys.forEach(key => {
+                tooltipArray.push(<p key={key}>{key + ': ' + this.props.tooltip[key]}</p>);
+            });
+        }
+        else {
+            console.log('Tooltip null');
+        }
+        return <React.Fragment>
+            {tooltipArray.map(element => {
+                return element;
+            })}
+        </React.Fragment>
+    }
+
+    render() {
+        return <div id='tooltip-box' ref={this.tooltipRef}>
```

Comment:
Avoid using a div ID, as it implies there is one and only one such div in the whole application, now and forever. Using a class would be better.